### PR TITLE
[Core/FilesystemExtensions] Add get list of directories method

### PR DESCRIPTION
**Pull request by sunethwarna**
**Description**
Adds a method in c++ to retrieve list of files and directories in a given path.

**Changelog**
Please summarize the changes in one list to generate the changelog:
E.g.
- Added `GetListOfDirectories`
- Added test

### CODE REVIEW

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp [hunk 1]
```diff
@@ -191,5 +191,53 @@ KRATOS_TEST_CASE_IN_SUITE(FileSystemRenameDirectory, KratosCoreFastSuite)
...
+    output_file.open(file_name_2);
+    KRATOS_CHECK(output_file.good());
+    output_file.close();
+
+    KRATOS_CHECK(Kratos::filesystem::exists(file_name_1));
+    KRATOS_CHECK(Kratos::filesystem::exists(file_name_2));
+    KRATOS_CHECK(Kratos::filesystem::exists(sub_dir));
+
+    const auto& list_of_dirs = Kratos::FilesystemExtensions::ListDirectory(dir_name);
+    const std::vector<std::string> check_list = {sub_dir, file_name_1, file_name_2};
```
Comment (philbucher) [inline kratos/tests/cpp_tests/sources/test_filesystem.cpp:231]: didn't you say on the phone you want to ignore subdirs?

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp [hunk 2]
```diff
@@ -191,5 +191,53 @@ KRATOS_TEST_CASE_IN_SUITE(FileSystemRenameDirectory, KratosCoreFastSuite)
...
+    KRATOS_CHECK(Kratos::filesystem::create_directory(sub_dir));
+
+    std::ofstream output_file;
+    output_file.open(file_name_1);
+    KRATOS_CHECK(output_file.good());
+    output_file.close();
+
+    output_file.open(file_name_2);
+    KRATOS_CHECK(output_file.good());
+    output_file.close();
```
Comment (philbucher) [inline kratos/tests/cpp_tests/sources/test_filesystem.cpp]: why do you check this with opening the file?
shouldn't the `exists` test be enough?

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp [hunk 3]
```diff
@@ -191,5 +191,53 @@ KRATOS_TEST_CASE_IN_SUITE(FileSystemRenameDirectory, KratosCoreFastSuite)
...
+    const std::string raw_file_name_1("dummy_file_1.txt");
+    const std::string raw_file_name_2("dummy_file_2.txt");
+    const std::string sub_dir(Kratos::FilesystemExtensions::JoinPaths({dir_name, dir_name_2}));
+    const std::string file_name_1(Kratos::FilesystemExtensions::JoinPaths({dir_name, raw_file_name_1}));
+    const std::string file_name_2(Kratos::FilesystemExtensions::JoinPaths({dir_name, raw_file_name_2}));
+
+    KRATOS_CHECK_IS_FALSE(Kratos::filesystem::exists(dir_name));
+    KRATOS_CHECK(Kratos::filesystem::create_directory(dir_name));
+    KRATOS_CHECK_IS_FALSE(Kratos::filesystem::exists(sub_dir));
+    KRATOS_CHECK(Kratos::filesystem::create_directory(sub_dir));
```
Comment (philbucher) [inline kratos/tests/cpp_tests/sources/test_filesystem.cpp]:
```suggestion
    KRATOS_CHECK_IS_FALSE(Kratos::filesystem::exists(sub_dir));
    KRATOS_CHECK(Kratos::filesystem::create_directories(sub_dir));
```

File: kratos/includes/kratos_filesystem.h [hunk 4]
```diff
@@ -57,8 +57,10 @@ namespace FilesystemExtensions {
 std::string KRATOS_API(KRATOS_CORE) CurrentWorkingDirectory();
 
 std::string KRATOS_API(KRATOS_CORE) JoinPaths(const std::vector<std::string>& rPaths);
+  
+std::vector<std::string> KRATOS_API(KRATOS_CORE) ListDirectory(const std::string& rPath);
```
Comment (philbucher) [inline kratos/includes/kratos_filesystem.h:82]: Can you add a short docstring
(I know I didn't for my methods, maybe you can add it there too? Cheers!)

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp (hunk 1 above)
Comment (sunethwarna) [reply kratos/tests/cpp_tests/sources/test_filesystem.cpp:231]: I was thinking like that, but then in python.listdir lists directories as well as files. So insteadt of having files only added directories as well so we won't be needing a seperate method to get subdirs. (is_directory and is_regular_file can be used to further identify files or directories)

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp (hunk 2 above)
Comment (sunethwarna) [reply kratos/tests/cpp_tests/sources/test_filesystem.cpp]: i will remove it... its a copy paste thing :/

File: kratos/includes/kratos_filesystem.h (hunk 4 above)
Comment (sunethwarna) [reply kratos/includes/kratos_filesystem.h:82]: I will add it

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp (hunk 1 above)
Comment (philbucher) [reply kratos/tests/cpp_tests/sources/test_filesystem.cpp:231]: ok makes sense to make it consistent :+1:

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp (hunk 2 above)
Comment (sunethwarna) [reply kratos/tests/cpp_tests/sources/test_filesystem.cpp]: done :)

File: kratos/tests/cpp_tests/sources/test_filesystem.cpp (hunk 3 above)
Comment (sunethwarna) [reply kratos/tests/cpp_tests/sources/test_filesystem.cpp]: done :)

File: kratos/includes/kratos_filesystem.h (hunk 4 above)
Comment (sunethwarna) [reply kratos/includes/kratos_filesystem.h:82]: done :)

**Review by philbucher — APPROVED** [review]

**Comment by philbucher** [comment]
cool!

**Pull request merged**